### PR TITLE
feat(grammar): add "inherits" field if available

### LINKED ISSUE #3294: Add parent language name to grammar.json
### Problem

There's no easy way to find the parent of a child grammar in tooling.

### Expected behavior

`tree-sitter generate` should add an "inherits" field to `grammar.json`.

### PR DESCRIPTION
**Pull request by ObserverOfTime**
The field can be excluded entirely by changing `null` to `undefined` if we want that.

Closes #3294.

### CODE REVIEW

File: cli/src/generate/dsl.js [hunk 1]
```diff
@@ -279,6 +282,14 @@ function grammar(baseGrammar, options) {
     throw new Error("Grammar's 'name' property must not start with a digit and cannot contain non-word characters.");
   }
 
+  if (inherits && typeof inherits !== "string") {
+    throw new Error("Grammar's 'inherits' property must be a string.");
```
Comment (amaanq) [inline cli/src/generate/dsl.js:286]:
```suggestion
    throw new Error("The inheriting grammar's 'name' property must be a string.");
```

File: cli/src/generate/dsl.js [hunk 2]
```diff
@@ -279,6 +282,14 @@ function grammar(baseGrammar, options) {
...
+  }
+
+  if (inherits && !/^[a-zA-Z_]\w*$/.test(name)) {
+    throw new Error("Grammar's 'inherits' property must not start with a digit and cannot contain non-word characters.");
```
Comment (amaanq) [inline cli/src/generate/dsl.js:290]:
```suggestion
    throw new Error("The inheriting grammar's 'name' property must not start with a digit and cannot contain non-word characters.");
```

Also allowing the name to start with digits might be okay

Comment (ObserverOfTime) [reply cli/src/generate/dsl.js:290]: It would not be a valid identifier (not sure if it matters here but better safe than sorry).

Comment (amaanq) [reply cli/src/generate/dsl.js]: yeah that's fine, if it is ever an issue we can adjust then

**Pull request merged**